### PR TITLE
Add RELEASE documentation

### PR DESCRIPTION
**Pull request by embano1**
Closes: #2368
Signed-off-by: Michael Gasch <mgasch@vmware.com>

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+```bash
+# Example
+git tag -a v0.25.0 -m "Release v0.25.0"
+```
+
+## Push the new Tag
+
+```bash
+# Will trigger Github Actions Release Workflow
+git push origin --tags
```
Comment (dougm) [inline RELEASE.md:33]: Maybe just the specific tag? `git push origin refs/tags/v0.25.0`

**Review by dougm — APPROVED** [review]
lgtm

File: RELEASE.md (hunk 1 above)
Comment (embano1) [reply RELEASE.md:33]: Good catch!

**Comment by embano1** [comment]
Addressed changes @dougm

**Review by dougm — APPROVED** [review]

**Pull request merged**